client/components/LearnerViewDirective.js

```diff
@@ -771,8 +804,14 @@ tie.directive('learnerView', [function() {
             var feedbackParagraphs = feedback.getParagraphs();
             var hasSyntaxError = false;
             for (var i = 0; i < feedbackParagraphs.length; i++) {
+              clearHighlight();
               if (feedbackParagraphs[i].isSyntaxErrorParagraph()) {
                 hasSyntaxError = true;
+                var errorContentArray = feedbackParagraphs[i].getContent()
+                  .split(' ');
+                var syntaxErrorLineNumber = errorContentArray[errorContentArray
```

Comment:
It is fine to add new methods, yes; make sure you test them too. The method could start by checking that the paragraph is of the appropriate type. and throwing an error if not.